include/model.h

```diff
@@ -28,21 +28,24 @@ class Model{
     void readFilePDB(const std::string&, bool);
 
     // export
+    bool createOutputFolder(std::string);
+    void createReport(std::string, std::vector<std::string>, bool);
+    void writeXYZfile(std::vector<std::tuple<std::string, double, double, double>>&, std::string);
     void writeSurfaceMap();
 
     std::vector<std::string> listElementsInStructure();
 
+    // crystal unit cell related functions
     bool getSymmetryElements(std::string, std::vector<int>&, std::vector<double>&);
-    bool createOutputFolder(std::string);
-    void createReport(std::string, std::vector<std::string>);
-    void writeXYZfile(std::vector<std::tuple<std::string, double, double, double>>&, std::string);
     bool processUnitCell(double, double, double, double);
     void orthogonalizeUnitCell();
     bool symmetrizeUnitCell();
     void moveAtomsInsideCell();
     void removeDuplicateAtoms();
+    void countAtomsInUnitCell();
     void generateSupercell(double);
     void generateUsefulAtomMapFromSupercell(double);
+    std::string generateUnitCellChemicalFormula(std::vector<std::string>);

```

Comment:
Why have a separate function for unit cells if the output is structurally the same?

Reply:
Because the data used is different (gui table vs map) and obtained from different places (Mainframe vs Model).
I will try to gather everything in controller.